llvm/lib/Transforms/Vectorize/SandboxVectorizer/SeedCollector.cpp

```diff
@@ -140,8 +140,8 @@ LLVM_DUMP_METHOD void SeedContainer::dump() const { print(dbgs()); }
 #endif // NDEBUG
 
 template <typename LoadOrStoreT> static bool isValidMemSeed(LoadOrStoreT *LSI) {
-  if (LSI->isSimple())
-    return true;
+  if (!LSI->isSimple())
```

Comment:
Is there any test for cases where the LoadOrStoreT is simple, but one of the later checks fails?

Reply:
added